core/src/dbs/session.rs

```diff
@@ -17,7 +17,7 @@ pub struct Session {
 	pub ip: Option<String>,
 	/// The current connection origin
 	pub or: Option<String>,
-	/// The current connection ID
+	/// The current user-defined connection ID
```

Comment:
We cannot hijack this variable. It is the WebSocket ID, not the session ID. There are multiple live queries per websocket. In the future we would likely want multiple sessions per websocket. Consequently, the session itself needs a separate ID field, and this should in the future be renamed to a websocket ID.

Reply:
Just out of curiosity, what do you man when you say that `session.id` does not identify a session? To me, session does not imply authentication, it implies that there is a state associated with some communication. The original use case for this was to be able to uniquely identify unauthenticated sessions. I would say we currently define a session as either a WS connection or the ephemeral session that is created to handle a single HTTP request that does not establish a persistent connection, regardless of whether or not authentication has happened.

Reply:
Thank you for the clarification! What I don't understand then is why you would want to switch to a different session within the same WebSocket connection. What would be the meaning of a session in that scenario? Would that only be useful to "clear" any live queries from the WebSocket connection? If so, I think we could probably design something more specific (e.g. a killall method) instead of overloading the concept of sessions.

It would be helpful to me if you can provide an example use case for the behavior that you would like to have. As I currently understand it, I would not add another session identifier, but there may be a strong case for it that I am missing. Let me know if I am misunderstanding!

Reply:
Got it, thanks a lot for the update! And yeah, absolutely agree on tracking authentication status of the session and flushing live queries being a desirable security guarantee. I mentioned `killall` just to describe the kind of operation that I understood you would expect. Currently, we [recommend ensuring that WebSocket connections are not reused between users](https://surrealdb.com/docs/surrealdb/reference-guide/security-best-practices#session-isolation), but this risk is something that it makes sense to try to prevent by default.

There is actually some work already going on for this, but I believe the solution in the works is intended to be more general and target both performance and security while ensure that this operation does not have a huge performance cost in cases there lots of live queries exist in the server.

With regards to this specific PR, I would leave the decision up to @phughk, as we may have some plans around live queries, sessions or the WS API that I am not aware of.